docs/funding.md

```diff
@@ -0,0 +1,53 @@
+# Announcing HTTPX 1.0
+
+HTTPX is a fully featured HTTP client for Python, which offers both HTTP/1.1 and HTTP/2 support. Learning from all of the functionality and ease of use that the fantastic `requests` library provides, while bringing additional features and keeping pace with HTTP as it evolves.
+
+* An API that is broadly compatible with `requests`.
+* Fully supports all the features that the `requests` package provides.
+* Both HTTP/1.1 and HTTP/2 support.
+* Strict request timeouts enabled by default.
+* Support for the asyncio and trio concurrency environments.
+* A fully type annotated interface, with 100% code coverage.
+* Development & maintenance that's backed by a sustainable and transparent business model.
+
+HTTPX is the **only** Python HTTP client with **HTTP/1.1 and HTTP/2** support, as well as being the only client providing both **sync and async** support. We'd like to continue pushing the project forward, and we're asking for sponsors to help us in building the foundation of modern HTTP stack for Python.
+
+We've spent a long time finessing the API for the project, and we're now happy to release a 1.0 version, and provide a standard SEMVER policy on future API updates.
+
+## Sponsorship plans
+
+We offer a number of sponsorship plans for businesses looking to support our collaborative development model.
+
+[Sign up as a sponsor](https://github.com/sponsors/encode){ .md-button }
+
+*Currently sponsorships are displayed primarily on the [Django REST framework](https://www.django-rest-framework.org) site, but we're looking at rolling these out across our other projects. Get in touch with your requirements, and we'll aim to work with you to ensure your sponsorship is properly advocated.*
+
+## Roadmap
+
+### HTTPX 1.1
+
+Released alongside a new version of `httpcore`, which will expose a lower-level API onto the core networking. This will allow users to dig further into the underlying behaviour, such as inspecting the state of the connection pool. Support for custom network transports will allow functionality such as record/replay to be elegantly implemented. This version will also add SOCKs proxy support.
+
+### HTTPX 1.2
+
+This version will add an integrated command line client to HTTPX, allowing for developers to easily switch between debugging HTTP requests from the console, and within their own codebases. You'll be able to switch easily between the two, with a consistent interface onto both sets of tools.
+
+### HTTPX 1.3 & beyond...
+
+Resource limits for the total download time, and maximum allowed download size. Support for WebSockets which run transparently over either HTTP/1.1 or HTTP/2. Documentation for using advanced functionality such as HTTP/2 bi-directional streaming, or working with network streams established using CONNECT or Upgrade requests. Support for a URLLib3 backend, which will aid projects that want upgrade from requests but still keep the underlying networking changes to an absolute minimum.
```

Comment:
websockets don't run over HTTP/1.1

Reply:
and they *can't* run over HTTP/2 https://daniel.haxx.se/blog/2016/06/15/no-websockets-over-http2/

Reply:
on HTTP 3 [EventSource](https://developer.mozilla.org/en-US/docs/Web/API/EventSource) should out-perform websockets

there also doesn't seem to be any interest in websockets over http3. I suspect because on HTTP 3 [EventSource](https://developer.mozilla.org/en-US/docs/Web/API/EventSource) should out-perform websockets.

The new developments are on https://datatracker.ietf.org/doc/html/draft-ietf-webtrans-http3 that adds "websocket"-like transports with multi homing and datagram support.

Reply:
```suggestion
Resource limits for the total download time, and maximum allowed download size. Support for WebSockets which run transparently over either HTTP/1.1 or HTTP/2. Documentation for using advanced functionality such as HTTP/2 bi-directional streaming, or working with network streams established using CONNECT or Upgrade requests. Support for a urllib3 backend, which will aid projects that want upgrade from requests but still keep the underlying networking changes to an absolute minimum.
```

Reply:
That 2016 blog post is out of date. The websockets-over-http/2 spec is RFC 8441.

Reply:
ah indeed it's shipping in chrome and firefox already https://www.chromestatus.com/feature/6251293127475200

Reply:
so I'm reading this as websockets can run on their own "compatible with http 1.1" using "http Upgrade" (as opposed to tunnelled 'over http 1.1'), RFC 8441 websockets however *are* tunnelled over HTTP 2 ?

Reply:
Well, it'd probably be a little clearer to say "negotiated transparently over HTTP/1.1 or HTTP/2".

The idea is that upgrading to a raw network read/write API would work equally well against either HTTP/1.1 or HTTP/2, without the developer seeing any difference.  Against HTTP/1.1 those read/write operations would map straight onto the TCP connection, but with HTTP/2 they'd map onto a single HTTP/2 stream.

We can achieve this without adding any extra API methods by exposing the stream as an extension.

```python
with client.get(..., headers={"Upgrade": ...}) as response:
    if response.status_code != 101:
        ...
    stream = response.extensions["stream"]
    ...
```

My feeling is that we probably want to tightly scope things, so that we *show* how HTTPX can be used at this kind of level of API, without necessarily also building a full websockets API on top of HTTPX. (We can leave that to other packages to build on top of.)